src/selection_criteria.rs

```diff
@@ -126,6 +126,30 @@ pub struct ReadPreferenceOptions {
     /// specified for an operation, the operation will return an error.
     #[builder(default)]
     pub max_staleness: Option<Duration>,
+
+    /// Specifies hedging behavior for reads.
+    ///
+    /// See the [MongoDB docs](https://docs.mongodb.com/manual/core/read-preference-hedge-option/) for more details.
+    #[builder(default)]
+    pub hedge: Option<HedgedRead>,
+}
+
+/// Specifies hedging behavior for reads.
+///
+/// See the [MongoDB docs](https://docs.mongodb.com/manual/core/read-preference-hedge-option/) for more details.
+#[derive(Clone, Debug, Deserialize, PartialEq, TypedBuilder)]
+#[non_exhaustive]
+pub struct HedgedRead {
+    /// Whether or not to allow reads from a sharded cluster to be "hedged" across two replica
+    /// set members per shard, with the results from the first response received back from either
+    /// being returned.
+    pub enabled: bool,
+}
+
+impl HedgedRead {
+    pub fn with_enabled(enabled: bool) -> Self {
```

Comment:
Given that there's currently only a single field and it's required, forcing users to do `HedgedRead::builder().enabled(true).build()` felt like a bit too much, so I added a constructor here. Given that any future fields added to the `hedge` document will not be required due to backwards compatibility, I think this is a safe API to add.

Reply:
I wonder if separate `HedgedRead::enabled` and `HeadgedRead::disabled` would be more clear. `HedgedRead::with_enabled(false)` just seems kind of confusing to me. `HedgedRead::with_enabled(true)` also seems a bit odd to me to be honest.

e.g.
```rust
let options = ReadPreferenceOptions::builder()
    .hedge(HedgedRead::with_enabled(false))
    .build();
// vs
let options = ReadPreferenceOptions::builder()
    .hedge(HedgedRead::disabled())
    .build();
```
and
```rust
let options = ReadPreferenceOptions::builder()
    .hedge(HedgedRead::with_enabled(true))
    .build();
// vs
let options = ReadPreferenceOptions::builder()
    .hedge(HedgedRead::enabled())
    .build();
```

Alternatively, we could go for `HedgedReadOptions` as @isabelatkinson suggests. In that case I think `with_enabled` is slightly less confusing, since it's more obvious that it's an option, albeit a required one.

Reply:
I'm not necessarily opposed to having separate constructors; I opted not to do that initially due to the fact that it might seem weird to call it "enabled" if other options exist later as well. It does seem like `HedgedReadOptions` is the best course, so I'll change it to that.